test/cpp/api/functional.cpp

```diff
@@ -73,6 +73,18 @@ TEST_F(FunctionalTest, CosineSimilarity) {
   ASSERT_TRUE(output.allclose(expected, 1e-04));
 }
 
+TEST_F(FunctionalTest, SmoothL1Loss) {
+  auto input = torch::tensor({{0.1}, {1.2}, {4.7}}, torch::requires_grad());
+  auto target = torch::tensor({{0.}, {1.}, {5.}}, torch::kFloat);
+  auto output =
+      F::smooth_l1_loss(input, target, SmoothL1LossOptions());
```

Comment:
We can remove `SmoothL1LossOptions()` in this line to test the "no options" use case :D